x-pack/plugins/maps/public/classes/styles/vector/components/legend/vector_style_legend.tsx

```diff
@@ -33,12 +34,18 @@ export function VectorStyleLegend({
   const legendRows = [];
 
   for (let i = 0; i < styles.length; i++) {
-    const row = styles[i].renderLegendDetailRow({
-      isLinesOnly,
-      isPointsOnly,
-      symbolId,
-      svg,
-    });
+    const styleMetaDataRequest = styles[i].isDynamic()
+      ? (styles[i] as DynamicStyleProperty<any>).getStyleMetaDataRequest()
+      : undefined;
+
+    const row = styleMetaDataRequest?.getError()
+      ? styleMetaDataRequest.renderError()
```

Comment:
I think we will need a component here because we still want to show the style property title with tooltip. You can look at `/Users/nathanreese/projects/kibana/x-pack/plugins/maps/public/classes/styles/vector/components/legend/breaked_legend.tsx` as an example, but you will want something like 

```
<div>
  <EuiFlexGroup gutterSize="xs" justifyContent="spaceBetween">
    <EuiFlexItem grow={false}>
      <EuiToolTip
        position="top"
        title={this.props.style.getDisplayStyleName()}
        content={this.state.label}
      >
        <EuiText className="eui-textTruncate" size="xs" style={{ maxWidth: '180px' }}>
          <small>
            <strong>{this.state.label}</strong>
          </small>
        </EuiText>
      </EuiToolTip>
    </EuiFlexItem>
  </EuiFlexGroup>
  <EuiFlexGroup direction="column" gutterSize="none">
    <EuiCallOut title="Unable to fetch style meta data" color="warning" iconType="warning">
      <p>
        {error.body}
      </p>
    </EuiCallOut>
  </EuiFlexGroup>
</div>
```